io/zenoh-link-commons/src/tcp.rs

```diff
@@ -60,6 +63,28 @@ impl<'a> TcpSocketConfig<'a> {
         dst_addr: &SocketAddr,
     ) -> ZResult<(TcpStream, SocketAddr, SocketAddr)> {
         let socket = self.socket_with_config(dst_addr)?;
```

Comment:
It seems that the socket will be bound twice here. Once in `socket_with_config` and once below.